hack/stabilization-changes.py

```diff
@@ -94,24 +95,33 @@ def stabilize_channel(name, channel, channels, channel_paths, **kwargs):
             **kwargs)
 
 
+def human_readable_relativedelta(delta):
+    attrs = ['years', 'months', 'days', 'hours', 'minutes']
+    human_readable = ['%d %s' % (getattr(delta, attr), attr if getattr(delta, attr) > 1 else attr[:-1])
+        for attr in attrs if getattr(delta, attr)]
+    return "{} ago".format(", ".join(human_readable))
+
+
 def stabilize_release(version, channel_name, channel_path, delay, errata, feeder_name, feeder_promotion, cache, waiting_notifications=True, **kwargs):
     now = datetime.datetime.now()
     version_delay = now - feeder_promotion['committer-time']
+    merged_ago = human_readable_relativedelta(
+        dateutil.relativedelta.relativedelta(now, feeder_promotion['committer-time']))
     errata_public = False
     public_errata_message = ''
+    commit_link = '<https://github.com/openshift/cincinnati-graph-data/commits/{0}|{0}>'.format(feeder_promotion['hash'][:10])
     if errata:
         errata_uri, errata_public = public_errata_uri(version=version, channel=feeder_name, cache=cache)
         if errata_uri:
             public_errata_message = ' {} is{} public.'.format(errata_uri, '' if errata_public else ' not')
     if (delay is not None and version_delay > delay) or errata_public:
         path_without_extension, _ = os.path.splitext(channel_path)
         subject = '{}: Promote {}'.format(path_without_extension, version)
-        body = 'It was promoted to the feeder {} by {} ({}, {}) {} ago.{}'.format(
+        body = 'It was promoted to the feeder {} by {} ({}) {}.{}'.format(
                 feeder_name,
-                feeder_promotion['hash'][:10],
-                feeder_promotion['summary'],
```

Comment:
This body is used for the commit message too:

```console
$ git --no-pager log --no-merges -1 channels/fast-4.*
commit 527f57a94f5fba118a5f256e9bbf2fb0e00f6fe2 (origin/promote-4.8.18-to-fast-4.9, origin/pr/1182)
Author: Automated Stabilization Robot <noreply@redhat.com>
Date:   Tue Nov 2 14:51:24 2021 +0000

    channels/fast-4.9: Promote 4.8.18
    
    It was promoted to the feeder candidate-4.9 by 885ef69e5e (Merge pull
    request #1177 from openshift/pr-candidate-4.8.18, 2021-10-28) 5 days,
    3:08:01.003849 ago. https://access.redhat.com/errata/RHBA-2021:4020 is
    public.
```

And while I don't mind HTTPS commit message links in the Slack notifications, I'd rather stick to the `%h (%s, %d)` format like:

```console
$ git --no-pager log --format=reference -1 885ef69e5e
885ef69 (Merge pull request #1177 from openshift/pr-candidate-4.8.18, 2021-10-28)
```

See [the kernel docs][1] (which do not include the date suggestion), [the Git docs][2], and git/git@1f0fc1db8599f87520494ca4f0e3c1b6fabdf997.

I'm open to formatting different bodies for Slack and the commit messages, but I'm also fine sticking with the current bias towards the read-forever commit messages, even if that makes the Slack messages a bit chatty.

[1]: https://www.kernel.org/doc/html/latest/process/submitting-patches.html#describe-your-changes
[2]: https://github.com/git/git/blob/0cddd84c9f3e9c3d793ec93034ef679335f35e49/Documentation/SubmittingPatches#L146-L167

Reply:
Do we need commit messages though? These are autogenerated and don't carry any additional useful info imo.

Reply:
The referenced PR in the merge-commit's subject seems useful to me.  And having the inlined message from the referenced commit just makes reading it friendlier than an abbreviated commit hash and date, which... might be a useful sanity check?  Also, a Slack-formatted link dropping in `<https://github.com/openshift/cincinnati-graph-data/commits/{0}|{0}>` seems like distracting noise in the commit message.

Reply:
Sure, added summary and merge date back